src/backends/loom_backend.py

```diff
@@ -588,13 +705,42 @@ def column_mutual_information(self, bdb, generator_id, modelnos, colnos0,
         server = self._get_preql_server(bdb, generator_id)
         target_set = server._cols_to_mask(server.encode_set(colnames0))
         query_set = server._cols_to_mask(server.encode_set(colnames1))
-        mi = server._query_server.mutual_information(
-            target_set,
-            query_set,
-            entropys=None,
-            sample_count=loom.preql.SAMPLE_COUNT
-        )
-        return mi
+        # Check if we have to marginalize the condition:
+        if self._marginize_cmi(constraints):
+            inner_numsamples = numsamples
+            conditioning_rows_loom_format = self._get_constraint_rows(
+                constraints,
+                bdb,
+                generator_id,
+                population_id,
+                modelnos,
+                server,
+                inner_numsamples
+            )
+
+        else: # Otherwise, no marginalization is needed.
+            conditioning_rows_loom_format = [
+                self._get_constraint_row(
+                    constraints,
+                    bdb,
+                    generator_id,
+                    population_id,
+                    server
+                )
+            ]
+        mi_estimates = [
+            server._query_server.mutual_information(
+                target_set,
+                query_set,
+                entropys=None,
+                sample_count=loom.preql.SAMPLE_COUNT, # XXX: wrong but quick;
```

Comment:
because the 1000 is very quick -- so practically, we won't need to control it. Thus, a user can use the `samples` parameter in BQL to control the number of times we call simulate for the marginalization through said `numsamples` parameter in BQL. The above choice will keep the code most usable for now. Eventually, we probably want to expose both parameters to BQL.